woocommerce/woocommerce-android
# Issue/3725 use backend feature flag for payments

Parent issue #3725 

~Please review https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2024 first~

Adds fetching of feature flag from the backend if site eligible for card-present payments. It's false by default.

It does the call in 2 cases:
* The app came to the foreground with a timeout of 10 mins
* The user changes a site

### How to test
Open Charles.

* Open the app and notice `payments/accounts` call. Hide the app and open again - a call should not be performed. Wait 10 mins and open the app. The call should be performed.
* Switch site in settings - call should be performed.
* Build release version. Notice that collect money buttons in order details and manages reader in settings are not visible. Kill the app and change Charles `card_present_eligible` flag to true. Notice that the aforementioned buttons are visible
